src/annotator/index.js

```diff
@@ -68,28 +68,32 @@ function init() {
     sidebar = new Sidebar(document.body, eventBus, guest, getConfig('sidebar'));
 
     // Expose sidebar window reference for use by same-origin guest frames.
-    window_.__hypothesis.sidebarWindow = sidebar.iframe.contentWindow;
+    window_.__hypothesis.sidebarWindow = sidebar.ready.then(
+      () => sidebar.iframe.contentWindow
+    );
   }
 
   // Clear `annotations` value from the notebook's config to prevent direct-linked
   // annotations from filtering the threads.
   const notebook = new Notebook(document.body, eventBus, getConfig('notebook'));
 
   // Set up communication between this host/guest frame and the sidebar frame.
-  const sidebarWindow = sidebar
-    ? sidebar.iframe.contentWindow
-    : /** @type {HypothesisWindow} */ (window.parent).__hypothesis
-        ?.sidebarWindow;
+  const sidebarWindow =
+    window_.__hypothesis.sidebarWindow ||
+    /** @type {HypothesisWindow} */ (window.parent).__hypothesis?.sidebarWindow;
```

Comment:
Currently only guest frames which direct children of the host frame and same-origin are supported. In this case `window.__hypothesis.sidebarWindow` is assigned in the host frame and is accessed via `window.parent.__hypothesis?.sidebarWindow` in the guest.

Reply:
Thanks for the clarification.

Reply:
Because the same code runs on the host frame and on the guest frame/s, it maybe  useful to add a comment or use a if/else indicating in which condition happens what assignment.

For example:
```
let sidebarWindow;
if (!annotatorConfig.subFrameIdentifier) {
// host
sidebarWindow = window.__hypothesis.sidebarWindow
} else {
// guest
sidebarWindow = window.parent.__hypothesis?.sidebarWindow
}
```

It is just a suggestion.